CodeModel/src/main/java/org/openzen/zenscript/codemodel/type/builtin/ArrayTypeSymbol.java

```diff
@@ -36,7 +36,7 @@ public class ArrayTypeSymbol implements TypeSymbol {
 
 	public static ArrayTypeSymbol get(int dimension) {
 		while (types.size() < dimension) {
-			types.add(new ArrayTypeSymbol(types.size()));
+			types.add(new ArrayTypeSymbol(types.size() + 1));
```

Comment:
Do we really need this caching here?

Reply:
Yes; each symbol is considered unique for both comparison and code generation. Generating new symbols every time would result in the same type being considered different.